jaxrs2/src/main/java/io/opentracing/contrib/jaxrs/client/SpanClientResponseFilter.java

```diff
@@ -0,0 +1,41 @@
+package io.opentracing.contrib.jaxrs.client;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Logger;
+
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientResponseContext;
+import javax.ws.rs.client.ClientResponseFilter;
+
+import io.opentracing.contrib.jaxrs.SpanWrapper;
+
+/**
+ * @author Pavol Loffay
+ */
+public class SpanClientResponseFilter implements ClientResponseFilter {
+
+    private static final Logger log = Logger.getLogger(SpanClientResponseFilter.class.getName());
+
+    private List<SpanDecorator> spanDecorators;
+
+    public SpanClientResponseFilter(List<SpanDecorator> spanDecorators) {
+        this.spanDecorators = spanDecorators;
```

Comment:
The usual approach is to copy the list when receiving it, and freeze it when sending it. So, your idea about freezing it when receiving the list is not bad, it's just not "usual", specially when you want to manipulate the list, without affecting the original list.